carmin.yaml

```diff
@@ -133,28 +332,147 @@ definitions:
       defaultStudy:
         type: string
+        format: ascii
       supportedAPIVersion:
         type: string
-  errorCodeAndMessage:
+        format: ascii
+    additionalProperties: true # allow extensions
+  Pipeline:
+    type: object
+    required:
+      - identifier
+      - name
+      - version
+      # - parameters optional in 0.2
+    properties:
+      identifier:
+        type: string
+        format: ascii
+      name:
+        type: string
+      version:
+        type: string
+        format: ascii
+      description:
+        type: string
+      canExecute:
+        type: boolean
+        description: true if the user who requested the pipeline can execute it
+      parameters:
+        type: array
+        items:
+          $ref: '#/definitions/PipelineParameter'
+      errorCodesAndMessages:
+        type: array
+        items:
+          $ref: '#/definitions/ErrorCodeAndMessage'
+  PipelineParameter:
+    type: object
+    required:
+      - name
+      - type
+      - isOptional
+      - isReturnedValue
+    properties:
+      name:
+        type: string
+        format: ascii
+      type:
+        $ref: '#/definitions/ParameterType'
+      isOptional:
+        type: boolean
+      isReturnedValue:
+        type: boolean
+      defaultValue:
+        description: Default value. It must be consistent with the parameter type.
+      description:
+        type: string
+  ParameterType:
+    type: string
+    enum: [File,String,Boolean,Int64,Double,List]
+  ErrorCodeAndMessage:
     type: object
     required:
       - code
       - message
     properties:
       code:
         type: integer
-        format: int32
+        format: int64
       message:
         type: string
-    
\ No newline at end of file
+  Execution:
+    type: object
+    required:
+      - name
+      - pipelineIdentifier
+      - inputValues
+    properties:
+      identifier:
+        type: string
+        format: ascii
+        readOnly: true
+        description: execution id. Must always be present in responses.
+      name:
+        type: string
+        description: execution name
+      pipelineIdentifier:
+        type: string
+        format: ascii
+        description: which pipeline that started this execution
+      timeout:
+        type: integer
+        format: int64
+        description: The timeout in seconds until which the execution is killed and deleted with all its files. 0 or absense means no timeout
+      status:
+        type: string
+        enum: [initializing,ready,running,finished,initialization_failed,execution_failed,unknown,killed]
+        readOnly: true
+        description: The status of the execution. Must always be present in responses.
+      inputValues:
+        type: object
+        additionalProperties: true
+        description:
+          Represents the input as a key/value object. The types should respect the parameters of the pipeline used for the execution.
+      returnedFiles:
```

Comment:
I think you misunterstood the specification because I tried to implement the solution 2 here.

"returnedFiles" is an json object where all values should be arrays. So a set of "key/list of string"' pairs.
I meant it to be like 

`
(...)
"returnedFiles" :{
"param1" : ["url1", "url2"],
"param2": ["url3"]
}
(...)`

Where "param1" and "param2" are output parameters.

Does that look OK ? I could describe it more in the description